src/include/btree_cmp_inline.h

```diff
@@ -14,13 +14,179 @@
 #endif
 #endif
 
-#if defined(HAVE_ARM_NEON_INTRIN_H)
-#include <arm_neon.h>
-#endif
-/* 16B alignment */
-#define WT_ALIGNED_16(p) (((uintptr_t)(p)&0x0f) == 0)
 #define WT_VECTOR_SIZE 16 /* chunk size */
 
+#ifdef HAVE_X86INTRIN_H
+/*
+ * __lex_compare_ge_16 --
+ *     Lexicographic comparison routine for data greater than or equal to 16 bytes. Returns: < 0 if
+ *     user_item is lexicographically < tree_item, = 0 if user_item is lexicographically =
+ *     tree_item, > 0 if user_item is lexicographically > tree_item. We use the names "user" and
+ *     "tree" so it's clear in the btree code which the application is looking at when we call its
+ *     comparison function. Some platforms like ARM offer dedicated instructions for reading 16
+ *     bytes at a time, allowing for faster comparisons.
+ */
+static WT_INLINE int
+__lex_compare_ge_16(const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp)
+{
+    __m128i res_eq, t, u;
+    uint64_t t64, tfirst, u64, ufirst;
+    const uint8_t *tendp, *treep, *uendp, *userp;
+    bool firsteq;
+
+    uendp = ustartp + len;
+    tendp = tstartp + len;
+
+    /* skip 16 matching bytes at a time, starting at first possible difference. */
```

Comment:
Since it's not obvious what the intrinsics are doing I'd suggest a blunter comment:
```suggestion
    /* Compare 16 bytes at a time until we find a difference or run out of 16 byte chunks to compare. */
```
Also in `__lex_compare_skip_ge_16`

Reply:
done